torch/_export/pass_infra/proxy_value.py

```diff
@@ -1,13 +1,12 @@
-# mypy: allow-untyped-defs
 # pyre-strict
 from typing import Union
-
+from typing import Iterator
 import torch
 
 
 class ProxyValue:
     # pyre-ignore
-    def __init__(self, data, proxy: Union[torch.fx.Proxy, torch.fx.Node]):
+    def __init__(self, data: Iterator[object], proxy: Union[torch.fx.Proxy, torch.fx.Node]):
```

Comment:
nit: you could use a type var to propogate the type from underlying self.data type if you wanted (and therefore preserve the type information.

```suggestion
    def __init__(self, data: Iterator[_T], proxy: Union[torch.fx.Proxy, torch.fx.Node]):
```

Reply:
Turns out this also takes in torch.Tensor. hurray typechecking!